client/components/domains/domain-search-results/index.jsx

```diff
@@ -77,33 +76,31 @@ var DomainSearchResults = React.createClass( {
 					cart={ this.props.cart }
 					onButtonClick={ this.props.onClickResult.bind( null, availableDomain ) } />
 				);
-		} else if ( this.props.suggestions && this.props.suggestions.length !== 0 && this.isDomainUnavailable() ) {
-			if ( this.props.products.domain_map && this.props.lastDomainError.code === 'not_available_but_mappable' ) {
-				const components = { a: <a href="#" onClick={ this.addMappingAndRedirect }/>, small: <small /> };
-				if ( this.props.domainsWithPlansOnly ) {
-					mappingOffer = this.translate( '{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}}' +
-						' with WordPress.com Premium.{{/small}}', {
-							args: { domain: lastDomainSearched },
-							components
-						}
-					);
-				} else {
-					if ( isNextDomainFree( this.props.cart ) ) {
-						mappingOffer = this.translate( '{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} for free.{{/small}}', {
-							args: {
-								domain: lastDomainSearched
-							},
-							components
-						} );
-					} else {
-						mappingOffer = this.translate( '{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} for %(cost)s.{{/small}}', {
-							args: {
-								domain: lastDomainSearched,
-								cost: this.props.products.domain_map.cost_display
-							},
-							components
-						} );
+		} else if ( suggestions && suggestions.length !== 0 && this.isDomainMappable() && this.props.products.domain_map ) {
```

Comment:
This is how it was in the original code, I've just pulled the `suggestions` from props to make the condition shorter. But I like your idea, one less check, more readable 👍